lib/stores/DataStore.js

```diff
@@ -37,13 +42,26 @@ class DataStore {
      *
      * http://tus.io/protocols/resumable-upload.html#creation
      *
-     * @param  {object} file File model.
+     * @param  {object} req http.incomingMessage
+     * @return {Promise}
      */
-    create(file) {
-        if (!(file instanceof File)) {
-            throw new Error(`${file} is not a File`);
-        }
-        console.log(`[DataStore] create: ${file.id}`);
+    create(req) {
+        console.log('[DataStore] create');
+
+        return new Promise((resolve, reject) => {
+            const upload_length = req.headers['upload-length'];
+            const upload_defer_length = req.headers['upload-defer-length'];
+            const upload_metadata = req.headers['upload-metadata'];
+
+            if (upload_length === undefined && upload_defer_length === undefined) {
+                return reject(412);
```

Comment:
These error code/messages should probably be constants, but I can do that in a separate PR :)